src/core/server/saved_objects/service/lib/errors.ts

```diff
@@ -162,6 +164,14 @@ export class SavedObjectsErrorHelpers {
     return isSavedObjectsClientError(error) && error[code] === CODE_CONFLICT;
   }
 
+  public static decorateEsCannotExecuteScriptError(error: Error, reason?: string) {
+    return decorate(error, CODE_ES_CANNOT_EXECUTE_SCRIPT, 501, reason);
```

Comment:
Using `501 Not Implemented` made the most sense to me when thinking about this. From [RFC 7231](https://tools.ietf.org/html/rfc7231#section-6.6.2):

>   **The 501 (Not Implemented) status code indicates that the server does
   not support the functionality required to fulfill the request.**  This
   is the appropriate response when the server does not recognize the
   request method and is not capable of supporting it for any resource.

Otherwise we could leave it as a 400 error (which we get from Elasticsearch when scripts cannot be executed), or use a 500 error.

Reply:
We don't really need a specific response code, we have the error type which is enough for the consumer. Updated in 4d75956, changed to a 400 error. I tried 500 but it swallows the error message.